### DISCUSSION TITLE
Thinking about the number of forwarding ports

### DISCUSSION THREAD

**Discussion by ls-2018** [skupperproject/skupper#1752] [Ideas]
For each port of each service, the router starts a port for forwarding. However, the number of ports that can be started by each pod is limited (the number of ports opened by a single pod port in the local cluster test is 1600+), how to face the situation of a large number of forwarding scenarios?

**Comment by ted-ross** [comment]
One possible way to handle scenarios where there are a large number of proxied services on one namespace is to deploy multiple router pods so as to split the "port-load" among them.

**Reply by ls-2018** [reply]
But each router pod starts one port, and each router pod can occupy a limited number of ports

**Reply by ted-ross** [reply]
I'm not sure I understand your comment.  If one router pod can open 1600 proxy ports, two pods could open 3200 and four could open 6400.  Since each pod has a different IP address, they each have their own independent port space.

**Comment by ls-2018** [comment]
<img width="1474" alt="image" src="https://github.com/user-attachments/assets/f799a2e5-da74-439d-b2d1-2276422fa22b">
<img width="1763" alt="image" src="https://github.com/user-attachments/assets/4ed28706-3cb4-4b15-a5a7-0dd9c153e3e4">
```
skupper expose deployment hello --target-namespace default --address asdsad --port 80
```

```
As can be seen from the figure, only one expose service router occupies at least one port.
When the number of services is increased, the number of ports that the router pod can run will reach its limit
```
